tests/test_gui.py

```diff
@@ -14,3 +16,41 @@ def test_gui_home(client: TestClient):
 
     assert response.status_code == 200
     assert "Select Directory" in response.text
+
+
+def test_gui_navigate_success(
+    client: TestClient,
+    tmp_path: Path,
+):
+    response = client.get("/?path=%s" % str(tmp_path))
+
+    assert response.status_code == 200
+
+
+def test_gui_navigate_not_found(
+    client: TestClient,
+    tmp_path: Path,
+):
+    response = client.get("/?path=%s" % str(tmp_path / "fake"))
+
+    assert response.status_code == 404
+    assert response.json() == {"detail": "Directory not found"}
+
+
+def test_gui_directory_selection(
+    client: TestClient,
+    tmp_path: Path,
+):
+    response = client.post("/directory_selection/", data={"directory": str(tmp_path)})
+
+    assert response.status_code == 200
+    assert response.json() == {"message": "You chose this directory: %s" % tmp_path}
+
+
+def test_gui_directory_selection_failure(
+    client: TestClient,
+    tmp_path: Path,
+):
+    response = client.post("/directory_selection/", data={"directory": str(tmp_path / "fake")})
+
+    assert response.status_code == 404
```

Comment:
Do you care to make an assertion about the response body here? I don't care either way, I just noticed that you did so for the `test_gui_navigate_not_found` case.

Reply:
updated in cbdc6e68684f30990e1b2f5e8764d59ac3fbbebf